### PR TITLE
Initial commit :rocket:

### PR DESCRIPTION
**Pull request by notlee**
See the README. Would love your feedback.

### CODE REVIEW

File: origami.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{
+	"description": "Maintain consistent spaces across components and projects in accordance to our baseline grid and design guidelines.",
+	"keywords": [],
```
Comment (JakeChampion) [inline origami.json]:
```suggestion
	"keywords": ["whitespace", "spacing"],
```

File: package.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+	"devDependencies": {
+		"eslint-config-origami-component": "^1.0.0"
```
Comment (JakeChampion) [inline package.json]: 👌

File: README.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+- [Contributing](#contributing)
+- [Migration](#migration)
+- [Contact](#contact)
+- [Licence](#licence)
+
+For [Build Service](https://www.ft.com/__origami/service/build/v2/) users, `o-spacing` provides [CSS classes](#markup) for vertical space and [CSS Custom Properties (CSS Variables)](#css-custom-properties) for other usecases. For Sass users `o-spacing` also provides a number of [Sass functions and mixins](#sass) for applying space to a project.
+
+### Spaces
+
+![named-spaces](https://user-images.githubusercontent.com/10405691/57918050-e9381780-788d-11e9-8310-ba5053c0c84a.png)
```
Comment (JakeChampion) [inline README.md]: Where does the original image live? Should it be in the repo?

Comment (notlee) [reply README.md]: It lives in the UXD Toolkit Sketch file. I'll add a link to it https://uxd.in.ft.com/standards/

Comment (notlee) [reply README.md]: Sorry linked to the wrong site before.

File: origami.json [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+	},
+	"supportStatus": "experimental",
+	"browserFeatures": {},
+	"ci": {
+		"circle": "https://circleci.com/api/v1/project/Financial-Times/o-spacing"
+	},
+	"demosDefaults": {
+		"sass": "demos/src/demo.scss",
+		"documentClasses": "o-typography-wrapper",
+		"dependencies": ["o-normalise", "o-typography", "o-colors"]
```
Comment (gvonkoss) [inline origami.json]: Small reminder to pin demo dependencies ✌️

Comment (notlee) [reply origami.json]: Oops, good catch 👍

File: README.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+| l24 | 96px |
+
+To apply named spaces see `o-spacing` [markup](#markup) and [Sass](#sass) documentation.
+
+### Markup
+
+`o-spacing` provides utility classes to help space Origami components vertically within your project.
+
+```html
+<!-- Add a 48px vertical margin between elements. -->
```
Comment (gvonkoss) [inline README.md]: Above _and_ below?

File: README.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+}
+```
+
+### Baseline Custom Property
+
+We recommend using a [named space](#named-spaces), but for more granular control a project may output any space which is a multiple of our baseline value. For users who do not have access to Sass, `o-spacing` outputs a `--o-spacing-baseline` CSS variable.
+
+```css
+.example {
+	padding: calc(var(--o-spacing-baseline) * 1); // A small padding.
```
Comment (gvonkoss) [inline README.md]:
```suggestion
	padding: calc(var(--o-spacing-baseline) * 1); // A small padding (4px).
```

File: README.md [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+	margin-bottom: calc(var(--o-spacing-baseline) * 4); // A large space.
```
Comment (gvonkoss) [inline README.md]:
```suggestion
	margin-bottom: calc(var(--o-spacing-baseline) * 4); // A large space (16px).
```

File: README.md [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+}
+```
+
+### Sass
+
+### Relative Units
+
+For compatibility with existing Origami projects, `o-spacing` outputs `px` values by default. To use relative `rem` values, set `$o-spacing-relative-units: true` before importing `o-spacing`.
```
Comment (gvonkoss) [inline README.md]: Could we include a code snippet as an example?

File: README.md [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+
+_If using `o-typography` set [$o-typography-relative-units](https://registry.origami.ft.com/components/o-typography@5.11.3/sassdoc?brand=master#variable-o-typography-relative-units) also._
+
+### Named Space
+
+We recommend Sass users apply space to their project using the `oSpaceByName` function. It accepts a [space name](#named-spaces) and returns a `px` value (or `rem` value, if [relative units](#relative-units) are enabled).
+
+```css
+	.example {
+		padding: oSpaceByName('s1');  // Small padding.
```
Comment (gvonkoss) [inline README.md]:
```suggestion
		padding: oSpaceByName('s1');  // Small padding (4px).
```

File: README.md [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+		margin-bottom: oSpaceByName('m12'); // Medium margin.
```
Comment (gvonkoss) [inline README.md]:
```suggestion
		margin-bottom: oSpaceByName('m12'); // Medium margin (48px).
```

File: README.md [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+	}
+```
+
+### Baseline Space
+
+We recommend users apply [named spaces](#named-space), but any space which is a multiple of our [baseline](#baseline) is allowed. To apply a multiple of the baseline value use `oSpaceByIncrement`. It accepts a increment to multiply the baseline value by and returns a `px` value (or `rem` value, if [relative units](#relative-units) are enabled).
```
Comment (gvonkoss) [inline README.md]:
```suggestion
We recommend the use of [named spaces](#named-space), but any space that multiplies our [baseline](#baseline) is allowed. To apply a multiple of the baseline value use `oSpaceByIncrement`. It accepts a value to multiply the baseline by and returns a `px` value (or `rem` value, if [relative units](#relative-units) are enabled).
```

File: demos/src/demo.scss [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+
+.demo-named-spaces-container {
+    padding-top: oSpaceByIncrement(3);
+    padding-left: oSpaceByIncrement(3);
+    background: repeating-linear-gradient(
+        0deg,
+        white,
+        white oSpaceByIncrement(1),
+        lightgrey oSpaceByIncrement(1),
+        lightgrey oSpaceByIncrement(2)
```
Comment (gvonkoss) [inline demos/src/demo.scss]: v meta 👌

File: src/scss/_mixins.scss [hunk 13]
```diff
@@ -0,0 +1,35 @@
+/// Output all o-spacing features.
```
Comment (gvonkoss) [inline src/scss/_mixins.scss]: All of the components that have a primary mixin set the primary mixin in the main.scss file, before calling it under `o-component-is-silent`. Could we move this over? 😬

File: README.md (hunk 5 above)
Comment (notlee) [reply README.md]: Just below, i.e. like typography headings

File: README.md [hunk 14]
```diff
@@ -0,0 +1,127 @@
+o-spacing [![Circle CI](https://circleci.com/gh/Financial-Times/o-spacing/tree/master.svg?style=svg)](https://circleci.com/gh/Financial-Times/o-spacing/tree/master)[![MIT licensed](https://img.shields.io/badge/license-MIT-blue.svg)](#licence)
+=================
+
+_Maintain consistent spaces across components and projects in accordance to our baseline grid and design guidelines._
+
+- [Markup](#markup)
+- [CSS Custom Properties](#css-custom-properties)
+- [Sass](#sass)
+- [Troubleshooting](#troubleshooting)
```
[the remark this answers is not in the document]
Comment (notlee) [reply README.md]: Good catch 👍

**Pull request merged**